neuralop/models/fno_block.py

```diff
@@ -1,10 +1,51 @@
 from torch import nn
 import torch.nn.functional as F
-
+import torch
 from .spectral_convolution import FactorizedSpectralConv
 from .skip_connections import skip_connection
 from .mlp import MLP
+import itertools
+import numpy as np
 
+def resample(x, res_scale, axis):
+    """
+    A module for generic n-dimentional interpolation (Fourier resampling).
+
+    Parameters
+    ----------
+    x : torch.Tensor
+            input activation of size (batch_size, channels, d1, ..., dN)
+    res_scale: int or tuple
+            Scaling factor along each of the dimensions in 'axis' parameter. If res_scale is scaler, then isotropic 
+            scaling is performed
+    axis: axis or dimensions along which interpolation will be performed. 
+    """
+    if isinstance(res_scale, (float, int)):
+        if isinstance(axis, (float, int)):
+            axis = [axis]
+            res_scale = [res_scale]
+        else:
+              res_scale = [res_scale]*len(axis)
+    else:
+        assert len(res_scale) == len(axis), "leght of res_scale and axis are not same"
+    index = np.argsort(axis)
+    axis = [axis[i] for i in index]
```

Comment:
What is the goal of these list comprehensions?

Reply:
this is to sort the axis and their corresponding scaling factors. For example, the parameters passed are
axis = [-1,-3,-2]
res_scale = [1,2,3]
After these lines, it will be 
axis  = [-3,-2,-1]
res_scale = [2,3,1]